File: proj_2/app/src/main/java/com/example/proj_2/tab2/InGalleryImageAdapter.java
package com.example.proj_2.tab2;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.proj_2.R;

import java.util.ArrayList;

public class InGalleryImageAdapter extends RecyclerView.Adapter<InGalleryImageAdapter.inGalleryImageHolder> {
    private ArrayList<MediaFileData> dataset;
    private Context context;

    InGalleryImageAdapter(ArrayList<MediaFileData> list){
        this.dataset = list;
    }

    public class inGalleryImageHolder extends RecyclerView.ViewHolder {
        public inGalleryImageHolder(View view) {
            super(view);
        }
    }

    @NonNull
    @Override
    public inGalleryImageHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View adapterLayout = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_image, parent, false);
        return new inGalleryImageHolder(adapterLayout);

    }

    @Override
    public void onBindViewHolder(@NonNull inGalleryImageHolder holder, int position) {
        MediaFileData item = dataset.get(position);
//        holder.itemView.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View v) {
//                Intent intent = new Intent(context, showFullImage.class);
//                intent.putExtra("uri", item.uri.toString());
//                context.startActivity(intent);
//
//
//                Glide.with(holder.itemView)
//                        .load(item.uri)
//                        .thumbnail(0.33f)
//                        .centerCrop()
//                        .into((ImageView) holder.itemView);
//            }
//        });


    }

    @Override
    public int getItemCount() {
        return dataset.size();
    }

}

File: sj/project1_final/app/src/main/java/com/example/project1_final/adapter/InGalleryImageAdapter.kt
package com.example.project1_final.adapter

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.project1_final.R
import com.example.project1_final.ShowFullImageActivity
import com.example.project1_final.model.MediaFileData

class InGalleryImageAdapter(private val context: Context, private val dataset: List<MediaFileData>):
        RecyclerView.Adapter<InGalleryImageAdapter.inGalleryImageHolder>() {
    class inGalleryImageHolder(private val view: View): RecyclerView.ViewHolder(view){
        val imageView: ImageView = view.findViewById(R.id.gridImg_infolder)
    }
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): inGalleryImageHolder {
        val adapterLayout = LayoutInflater.from(parent.context).inflate(R.layout.item_image, parent, false)
        return inGalleryImageHolder(adapterLayout)
    }


    override fun onBindViewHolder(holder: inGalleryImageHolder, position: Int) {
        val item = dataset.getOrNull(position)

        //holder.gridSub.text = item!!.dateTaken.toString()

        holder.imageView.setOnClickListener{
            val intent: Intent = Intent(context, ShowFullImageActivity::class.java)
            intent.putExtra("uri", item!!.uri.toString())
            context.startActivity(intent)
        }

        Glide.with(holder.imageView)
                .load(item!!.uri)
                .thumbnail(0.33f)
                .centerCrop()
                .into(holder.imageView)
    }

    override fun getItemCount(): Int {
        return dataset.size
    }
}